src/components/Forms/MultiAmountSelector.vue

```diff
@@ -0,0 +1,289 @@
+		>
+			<input
+				class="radio"
+				type="radio"
+				:id="`${id}-${option.key}`"
+				:disabled="option.disabled"
+				:value="option.key"
+				v-model="checked"
+				@change="onChange(option.key)"
+			>
+			<label
+				:class="{ 'label': true, 'split-pill-label': splitPills }"
+				:for="`${id}-${option.key}`"
+			>
+				{{ option.title }}
+			</label>
+			<transition name="kvfade">
+				<div
+					class="custom-amount-holder"
+					v-if="selected === 'custom' && option.key === 'custom'"
+				>
+					<kv-currency-input
+						id="custom-amount-input"
+						class="input-element"
+						:class="{'custom-input-element': true, 'error': $v.$invalid }"
+						v-model="customAmountModel"
+						:aria-disabled="!selected === 'custom'"
+						:disabled="!selected === 'custom'"
+					/>
+				</div>
+			</transition>
+		</div>
+		<ul class="validation-errors" v-if="$v.$invalid">
+			<li v-if="!$v.customAmountModel.minValue || !$v.customAmountModel.maxValue">
+				Enter an amount of $5-$10,000
+			</li>
+		</ul>
+	</div>
+</template>
+
+<script>
+import { validationMixin } from 'vuelidate';
+import { required, minValue, maxValue } from 'vuelidate/lib/validators';
+import KvCurrencyInput from '@/components/Kv/KvCurrencyInput';
+
+export default {
+	components: {
+		KvCurrencyInput
+	},
+	mixins: [
+		validationMixin
+	],
+	props: {
+		id: {
+			type: String,
+			required: true
+		},
+		options: {
+			type: Array,
+			required: true,
+			default() {
+				return [
+					{
+						title: '',
+						key: '',
+						disabled: false
+					}
+				];
+			}
+		},
+		/**
+     	* A key from options array you want to be selected
+     	*/
+		selected: {
+			type: String,
+			default: '',
+		},
+		splitPills: {
+			type: Boolean,
+			default: false,
+		},
+		customAmount: {
+			type: Number,
+			default: 25,
+		},
+		minCustomAmount: {
+			type: Number,
+			default: 25,
+		},
+		maxCustomAmount: {
+			type: Number,
+			default: 10000,
+		}
+	},
+	data() {
+		return {
+			checked: this.selected,
+			customAmountModel: this.customAmount,
+		};
+	},
+	validations() {
+		return {
+			customAmountModel: {
+				required,
+				minValue: minValue(this.minCustomAmount),
+				maxValue: maxValue(this.maxCustomAmount)
+			},
+		};
+	},
+	watch: {
+		customAmountModel(newVal) {
+			// emit a custom amount updated event only if valid
+			if (!this.$v.$invalid) {
+				// Implementation Required in Parent Component
+				// Enables tracking of custom option/key associated value
+				this.$emit('custom-amount-updated', newVal);
+			}
+		},
+		selected(newVal) {
+			// to prevent unnecessary validations from showing
+			if (newVal !== 'custom' && this.$v.$invalid) {
+				// reset to valid default value
+				this.customAmountModel = this.minCustomAmount;
+			}
+		},
+	},
+	methods: {
+		onChange(key) {
+			// Implementation Required in Parent Component
+			// Enables tracking of selected option/key
+			this.$emit('pill-toggled', key);
```

Comment:
We can get Storybook to document our events if we comment with JsDoc. There's an example [here](https://github.com/kiva/ui/blob/master/src/components/Kv/KvCheckbox.vue#L53)